diff --git a/src/misc.cpp b/src/misc.cpp
index c75f2ec..de44554 100644
--- a/src/misc.cpp
+++ b/src/misc.cpp
@@ -140,17 +140,10 @@ void dbg_print() {
 
 /// Used to serialize access to std::cout to avoid multiple threads writing at
 /// the same time.
-
+#include <jstockfish.h>
 std::ostream& operator<<(std::ostream& os, SyncCout sc) {
-
-  static Mutex m;
-
-  if (sc == IO_LOCK)
-      m.lock();
-
-  if (sc == IO_UNLOCK)
-      m.unlock();
-
+  if (sc == IO_LOCK) jstockfish::io_lock();
+  if (sc == IO_UNLOCK) jstockfish::io_unlock();
   return os;
 }
 
diff --git a/src/misc.h b/src/misc.h
index 37f1e70..de8e38e 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -57,7 +57,7 @@ enum SyncCout { IO_LOCK, IO_UNLOCK };
 std::ostream& operator<<(std::ostream&, SyncCout);
 
 #define sync_cout std::cout << IO_LOCK
-#define sync_endl std::endl << IO_UNLOCK
+#define sync_endl IO_UNLOCK
 
 
 /// xorshift64star Pseudo-Random Number Generator
